File: programmers/etc/level1/소수_찾기.py
def solution(n):
    sieve = [True]*1000001
    sieve[0] = sieve[1] = False
    for i in range(2, 1001):
        if sieve[i]:
            for j in range(2*i, 1000001, i):
                sieve[j] = False
    return sieve[:n+1].count(True)


n = 10
print(solution(n)) # 4

n = 5
print(solution(n)) # 3

File: Java/programmers/etc/level1/소수찾기.java
package CodeTest.Java.programmers.etc.level1;

import java.util.*;

public class 소수찾기 {
    public int solution(int n) {
        boolean[] sieve = new boolean[1000001];
        Arrays.fill(sieve, true);
        sieve[0] = sieve[1] = false;
        
        for (int i=2; i<1001; i++) {
            if (sieve[i]) {
                for (int j=2*i; j<1000001; j+=i) {
                    sieve[j] = false;
                }
            }
        }
        
        int answer = 0;
        for (int i=0; i<=n; i++) {
            if (sieve[i]) answer++;
        }
        
        return answer;
    }
}
